### PR TITLE
zig: new package at 0.7.1

### PR DESCRIPTION
**Pull request by Ominitay**
## Description of package
Zig is a general-purpose programming language.

## Installed manifest of package
```
It's too long -- GitHub won't let me post it.
```

## New package

- [x] Latest upstream version.
- [x] I agree to maintain this package.
- [x] I agree to notify the @kiss-community maintainers if I can no longer maintain this package.

### CODE REVIEW

**Comment by Ominitay** [comment]
This package requires that the `LLVM` package is modified. The line `-DLLVM_TARGETS_TO_BUILD="host;AMDGPU" \` in the `LLVM` build script must either be removed or changed to `-DLLVM_TARGETS_TO_BUILD="all" \`.

File: community/zig/build [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+mkdir -p build
+cd build
+
+cmake \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_INSTALL_PREFIX=/usr \
+    ..
+
+make DESTDIR="$1" install
```
Comment (illiliti) [inline community/zig/build:11]: https://k1sslinux.org/wiki/kiss/style-guide#0800

```
export DESTDIR="$1"

cmake -B build \
    -DCMAKE_BUILD_TYPE=Release \
    -DCMAKE_INSTALL_PREFIX=/usr 

cmake --build   build
cmake --install build
```

Comment (dilyn-corner) [reply community/zig/build:11]: Swap the `PREFIX` and `BUILD_TYPE` :)

**Comment by dilyn-corner** [comment]
In general for large manifests you can use a pastebin (in fact I encourage their usage - scrolling PRs on my phone becomes a chore)

**Comment by Ominitay** [comment]
@dilyn-corner Here is a paste :)
https://termbin.com/vq7w

**Comment by Ominitay** [comment]
Why hello again ;)

Changes will be pushed soon. I'm just checking that things all build right.

**Comment by Ominitay** [comment]
Everything is set now! This will build fine leaving LLVM unmodified. All that is left now is for `lld` to be merged first.

**Pull request merged**